sklearn/feature_extraction/text.py

```diff
@@ -1723,8 +1723,7 @@ def transform(self, X, copy=True):
             # name:
             check_is_fitted(self, attributes=["idf_"], msg="idf vector is not fitted")
 
-            # *= doesn't work
-            X = X * self._idf_diag
+            X = X @ self._idf_diag
```

Comment:
I make a change here. We relied on the fact that we have matrices at this point and the `*` operator provides a matrix multiplication.

We can solve it by always using `@` that is always the matrix multiplication whether you are dealing with arrays or matrices.

Reply:
Thanks for the explanation!